### PR TITLE
[#723] Fix TypeProvider expectations for Java 17 Enum style 

### PR DESCRIPTION
**Pull request by cdietrich**
[#723] Fix TypeProvider expectations for Java 17 Enum style

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.1.1-bin.zip
```
Comment (cdietrich) [inline gradle/wrapper/gradle-wrapper.properties:3]: this change is from separate PR and needed to run with java 17

File: org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java [hunk 2]
```diff
@@ -819,24 +819,46 @@ public void testMemberCount_15() {
...
-		assertEquals(2, constructorCount);
-		int fieldCount = TestEnum.class.getDeclaredFields().length;
-		// FirstValue, SecondValue, string, ENUM$VALUES
-		assertEquals(Arrays.toString(TestEnum.class.getDeclaredFields()), 4, fieldCount);
-		// ENUM$VALUES is synthetic
-		// TestEnum(String, String, EnumType) is synthetic
-		assertEquals(type.getMembers().toString(), innerTypesCount + methodCount + constructorCount + fieldCount - 2,
-				type.getMembers().size());
+		try {
+			String typeName = TestEnum.class.getName();
```
Comment (cdietrich) [inline org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java:823]: am not sure what our common pattern in extras is to deal with different java versions

Comment (szarnekow) [reply org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java:823]: I don't think we do have a pattern for that.
How did you like the assumeTrue(..) thingy?

Comment (cdietrich) [reply org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java:823]: we have the old variant. will apply it here

Comment (cdietrich) [reply org.eclipse.xtext.common.types.tests/tests/org/eclipse/xtext/common/types/access/impl/AbstractTypeProviderTest.java:823]: problem is eclipse. the enum is compiled differently there as of now

**Pull request merged**